Mathlib/Data/Matrix/Rank.lean

```diff
@@ -274,4 +275,35 @@ theorem rank_eq_finrank_span_row [LinearOrderedField R] [Finite m] (A : Matrix m
   rw [← rank_transpose, rank_eq_finrank_span_cols, transpose_transpose]
 #align matrix.rank_eq_finrank_span_row Matrix.rank_eq_finrank_span_row
 
+section MulNonsingInv
+
+/-! ### Lemmas about pre or post multiplying by an invertible matrix. -/
+
+/-- Pre (Left) multiplying by an invertible matrix does not change the rank -/
+lemma rank_mul_IsUnit {m n R: Type} [Fintype n] [DecidableEq n] [CommRing R]
+    (A: Matrix n n R) (B: Matrix m n R) (hA: IsUnit A.det) :
+    (B ⬝ A).rank = B.rank := by
+  rw [Matrix.rank, mulVecLin_mul, LinearMap.range_comp_of_range_eq_top, ←Matrix.rank ]
+  rw [LinearMap.range_eq_top] -- On separate line since implicit rfl works in prev line
+  intro x
+  use ((A⁻¹).mulVecLin x)
+  rwa [mulVecLin_apply, mulVecLin_apply, mulVec_mulVec, mul_nonsing_inv, one_mulVec]
+
+/-- Post (right) multiplying by an invertible matrix does not change the rank -/
+lemma rank_IsUnit_mul {m n R: Type} [Fintype m] [Fintype n] [DecidableEq m] [Field R]
```

Comment:
Maybe
```suggestion
lemma rank_mul_eq_right_of_isUnit {m n R: Type} [Fintype m] [Fintype n] [DecidableEq m] [Field R]
```
to match `rank_mul_le_right`?

Reply:
I don't think that is a good name.  I guess what I consider a better name is something like [`Matrix.rank_conjTranspose`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Matrix/Rank.html#Matrix.rank_conjTranspose) or [`Matrix.rank_self_mul_conjTranspose`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Matrix/Rank.html#Matrix.rank_self_mul_conjTranspose).

Both names describe the LHS and probably rely on you implicitly knowing the RHS (I am guessing here).

Reply:
> places [...] the word isUnit [...] to the RHS of eq

This is by design; A lemma that concluses `C` from `H` should be called `$(name C)_of_$(name H)$` (or just `name C` if H is obvious)

Neither of the examples you give has such a hypothesis.

Reply:
But the conclusion here is something of nature rank_...._eq_rank. Maybe
`rank_mul_isUnit_eq_rank

Reply:
The conclusion here would be called `rank_mul_eq_right` or `rank_mul_eq`. `IsUnit` is not part of the conclusion, it's part of the hypothesis: so it belongs after an `_of_`.